src/Command/PurgeCommand.php

```diff
@@ -0,0 +1,249 @@
+use Psr\Log\LogLevel;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Exception\InvalidArgumentException;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Question\ChoiceQuestion;
+use Symfony\Component\Filesystem\Filesystem;
+use Terminal42\LeadsBundle\Leads;
+use \Exception;
+
+class PurgeCommand extends Command
+{
+    /**
+     * @var ContaoFrameworkInterface
+     */
+    private $framework;
+
+    /**
+     * @var Connection
+     */
+    private $db;
+
+    /**
+     * @var Filesystem
+     */
+    private $fs;
+
+    /**
+     * Constructor.
+     *
+     * @param ContaoFrameworkInterface $framework
+     * @param Connection $db
+     * @param Filesystem $fs
+     */
+    public function __construct(ContaoFrameworkInterface $framework, Connection $db, Filesystem $fs = null)
+    {
+        if (null === $fs) {
+            $fs = new Filesystem();
+        }
+
+        $this->framework = $framework;
+        $this->db = $db;
+        $this->fs = $fs;
+
+        parent::__construct();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function configure()
+    {
+        $this
+            ->setName('leads:purge')
+            ->setDescription('Purge the leads outsite the configured storage period.');
+    }
+
+    /**
+     * @param InputInterface $input
+     * @param OutputInterface $output
+     * @return int|null|void
+     * @throws \Doctrine\DBAL\DBALException
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $this->framework->initialize();
+
+        $purged = $this->executeBatchPurge();
+
+        if ($purged) {
+            $logMessage = 'The leads have been purged successfully.';
+        } else {
+            $logMessage = 'No leads to purge.';
+        }
+
+        $logLevel = LogLevel::INFO;
+        $logger = System::getContainer()->get('monolog.logger.contao');
+        $logger->log($logLevel, $logMessage, array('contao' => new ContaoContext(__METHOD__, $logLevel)));
+        $output->writeln('<info>'.$logMessage.'</info>');
+    }
+
+    /**
+     * @return bool
+     * @throws \Doctrine\DBAL\DBALException
+     */
+    private function executeBatchPurge()
+    {
+        $purged = false;
+        $forms = $this->db->fetchAll("SELECT id, title, leadPeriod FROM tl_form WHERE leadPeriod > 0");
+
+        foreach ($forms as $masterForm) {
+
+            if (!empty($leads = $this->getAllLeads($masterForm))) {
+                $leadsIds = implode(',', array_keys($leads));
+
+                $deletedUploads = null;
+                if (!empty($leadsData = $this->getAllLeadsData($masterForm, $leadsIds))) {
+                    $leadsDataIds = implode(',', array_keys($leadsData));
+                    $deletedData = $this->db->executeUpdate(
```

Comment:
If I remember correctly, there is a `$this->db->delete(…)` method available that could simplify this?